### DISCUSSION TITLE
Is there a way to type the `Result` ?

### DISCUSSION THREAD

**Discussion by odcey** [wevm/wagmi#885] [Q&A]
Example from documentation
`useContractRead` 
```ts
{
  data?: Result
  error?: Error
  isIdle: boolean
 ...
  }) => Promise<Result>
  status: 'idle' | 'error' | 'loading' | 'success'
}
```

I would love to be able to type `Result` (for example to `BigNumber`, `string`, etc)
I used to use react-query `useQuery` and then do contract call inside with the help of generated typechains with hardhat.
What was great with this method was that the types were kept.
Here (maybe i'm wrong) but there's no way to pass/override the type of the `Result`.  

Example of what i was doing before using wagmi methods
```ts
const typedContract = Test__factory.connect("address", web3Provider);
const result = await typedContract.method();
// Here the type of result was automatically typed to BigNumber | string or other types based on typechains
```
I can still use this method but then i'll lose the interest of all the methods implemented by wagmi

**Comment by tmm** [comment]
Type inference based on ABIs is coming once I get back from some time off

https://twitter.com/awkweb/status/1555678944770367493
